File: client/src/scripts/Entity.ts
import { FloatingCoord } from "./Coord";
import { ItemType } from "./Cell";

export enum EntityType {
  Snake = 14,
  Dog = 15,
  Rat = 16,
}

export class Entity {
  public coord: FloatingCoord;
  public name: string = null;
  public spriteId = 6;
  public pv: number;
  public maxPv: number;
  public inventory: ItemType[];
  public coolDownAttack: number;
  public coolDownDash: number
  public isDashing: boolean;
}

File: server/rogue-like-multi-server/Entity.cs
using System.Collections.Generic;
using System.Linq;
using Newtonsoft.Json;

namespace rogue_like_multi_server
{
    public class Entity
    {
        [JsonProperty("coord")]
        public Coord Coord;

        [JsonProperty("name")]
        public string Name;

        [JsonProperty("spriteId")]
        public int SpriteId;

        [JsonProperty("inventory")]
        public IList<ItemType> Inventory;

        [JsonProperty("pv")]
        public int Pv;

        [JsonProperty("maxPv")]
        public int MaxPv;

        public Entity(Coord coord, string name, int spriteId, IList<ItemType> inventory, int maxPv)
        {
            Coord = coord;
            Name = name;
            SpriteId = spriteId;
            Inventory = inventory;
            Pv = maxPv;
            MaxPv = maxPv;
        }

    }
}